File: math_stuff/less_interesting_languages/triangle.py
# run with `python triangle.py`

def triangle(num: int) -> int:
    return sum(range(num+1))

def pyramid(num: int) -> int:
    return sum(map(lambda x: x*x, range(num+1)))

def main():
    print("triangle numbers")
    print([triangle(i) for i in range(10+1)])
    print("pyramid numbers")
    print([pyramid(i) for i in range(10+1)])

if __name__ == "__main__":
    main()

File: math_stuff/less_interesting_languages/triangle.go
package main
// run with `go run triangle.go`

import "fmt"

func triangle(n int) int {
	r := 0
	for i := 1; i <= n; i++ {
		r += i
	}
	return r
}

func pyramid(n int) int {
	r := 0
	for i := 1; i <= n; i++ {
		r += i * i
	}
	return r
}

func main() {
	fmt.Println("triangle numbers")
	for i := 1; i <= 10; i++ {
		fmt.Print(triangle(i), " ")
	}
	fmt.Println()
	fmt.Println("pyramid numbers")
	for i := 1; i <= 10; i++ {
		fmt.Print(pyramid(i), " ")
	}
	fmt.Println()
}
